satpy/composites/__init__.py

```diff
@@ -1477,6 +1460,44 @@ def _set_data_nans(self, data, mask, attrs):
 
         return data
 
+    def _select_data_bands(self, data_in):
+        """Select data to be composited from input data.
+
+        From input data, select the bands that need to have masking applied.
+        """
+        if 'bands' in data_in.dims:
+            return [data_in.sel(bands=b) for b in data_in['bands'] if b != 'A']
+        if self.mode == "RGBA":
```

Comment:
Should an error be raised if `self.mode` is neither `LA` nor `RGBA` ?

Reply:
I don't know.  Should it?  It's an internal pseudo-private function.  If we check `mode` for being supported, it would seem that this is not the best place to do so — I'd rather do such checks at the beginning of the compositor call.

Reply:
Well, what happens if the user passes a mode like YCbCrA and nothing happens? bug hunting for a few hours? ;)

Reply:
Exactly the same will happen as today.  I'm all for checking, but this seems to be the wrong place.

Reply:
Added a check when the compositor is created.  See
https://github.com/pytroll/satpy/pull/2005/commits/a115e91b597c337734f154557a730346b71dd94b

Reply:
Looks great, thanks!